src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/Validators/ValidationBehavior.shared.cs

```diff
@@ -291,12 +296,28 @@ async ValueTask UpdateStateAsync(bool isForced, CancellationToken? parentToken =
 
 		void UpdateStyle()
 		{
-			if (View == null || (ValidStyle ?? InvalidStyle) == null)
+			if (View == null)
+			{
 				return;
+			}
 
-			View.Style = IsValid
-				? ValidStyle
-				: InvalidStyle;
+			if ((ValidStyle ?? InvalidStyle) != null)
+			{
+				View.Style = IsValid
+					? ValidStyle
+					: InvalidStyle;
+				return;     // either use VSM or styles?
+			}
+
+			var vsmGroupsList = VisualStateManager.GetVisualStateGroups(View);
+			if (vsmGroupsList.Count > 0)
+			{
+				var vsmGroupsstates = vsmGroupsList[0].States;
+				if (vsmGroupsstates.All(x => x.Name.Equals(validVisualState) || x.Name.Equals(invalidVisualState)))
+				{
+					VisualStateManager.GoToState(View, IsValid ? validVisualState : invalidVisualState);
+				}
+			}
 		}
```

Comment:
```csharp
		void UpdateStyle()
		{
			if (View == null)
				return;

			VisualStateManager.GoToState(View, IsValid ? validVisualState : invalidVisualState);

			if ((ValidStyle ?? InvalidStyle) == null)
				return;

			View.Style = IsValid
				? ValidStyle
				: InvalidStyle;
		}
```

Reply:
Checking is not required ?

Reply:
I am not an expert in VisualStateManager)
So, I guess it's not required.

CC @pictos @jfversluis

Reply:
Sorry, checking what exactly?

Reply:
@jfversluis Checking if dev has defined a VisualStateGroup with at least a VisualState having as name "Valid" or Invalid, or it is not require to check before invoking `VisualStateManager.GoToState(View, IsValid ? validVisualState : invalidVisualState);`

Reply:
If Style accepts `null` I believe that we don't the check. But the best case is to do a test and see how it goes.

Reply:
@Cfun1 just to double-check, the current implementation works as expected with null values?